### PR TITLE
Cont. Check constraint

### PR DESCRIPTION
**Pull request by billy1624**
## PR Info

- Closes https://github.com/SeaQL/sea-query/issues/397
- Continue https://github.com/SeaQL/sea-query/pull/542

## New Features

- [x] Create check constraint on column definition and table
- [x] Add check constraint on table alter statement

### CODE REVIEW

**Comment by billy1624** [comment]
Testing on live db:
- https://dbfiddle.uk/WlH4QiGh
- https://dbfiddle.uk/m02fr8dO
- https://dbfiddle.uk/rXmp6eAP
- https://dbfiddle.uk/JEWq2nYq

**Comment by billy1624** [comment]
Hey @LemarAb, please check :D

File: src/backend/table_builder.rs [hunk 1]
```diff
@@ -46,6 +46,14 @@ pub trait TableBuilder:
             count += 1;
         }
 
+        for check in create.check.iter() {
```
Comment (ikrivosheev) [inline src/backend/table_builder.rs:49]: @billy1624 maybe is better use `create.check.iter().fold()`, like in other places in `src/backend`?

**Review by ikrivosheev — COMMENTED** [review]
@billy1624 thank you! One comment (question).

File: src/backend/table_builder.rs (hunk 1 above)
Comment (billy1624) [reply src/backend/table_builder.rs]: Good call. Please check https://github.com/SeaQL/sea-query/pull/567/commits/8b9a1d93bc2f0a41e78d5fa9c4112e85a1f863f5. I rewrite the whole part.

Comment (ikrivosheev) [reply src/backend/table_builder.rs]: Hmmm, I mean:
```
.fold(true, |first, _| {
    ...
    false
}
```

Comment (billy1624) [reply src/backend/table_builder.rs]: I like the the `for_each` version over the `fold` version

- `for_each`:
    ```rs
    let mut first = true;

    create.columns.iter().for_each(|column_def| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_column_def(column_def, sql);
        first = false;
    });

    create.indexes.iter().for_each(|index| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_table_index_expression(index, sql);
        first = false;
    });
    ```
- `fold`:
    ```rs
    let mut first = true;

    first = create.columns.iter().fold(first, |first, column_def| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_column_def(column_def, sql);
        false
    });

    first = create.indexes.iter().fold(first, |first, index| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_table_index_expression(index, sql);
        false
    });
    ```

Comment (ikrivosheev) [reply src/backend/table_builder.rs]: @billy1624 thank you for explanation, yes, it looks better.

**Pull request merged**